pkg/controller/vsphere/reconciler.go

```
@@ -233,6 +233,11 @@ func (r *Reconciler) reconcileMachineWithCloudState(vm *virtualMachine, taskRef
 		return err
 	}
 
+	klog.V(3).Infof("%v: reconciling powerstate annotation", r.machine.GetName())
+	if err := r.reconcilePowerStateAnnontation(vm); err != nil {
+		klog.V(3).Infof("%s: Failed to get power state (machine may be being deleted): %v", r.machine.Name, err)
```

why are we ignoring the error here? we should return an error.

---

shouldn't we?